src/scalaDebugger.ts

```diff
@@ -57,6 +60,37 @@ class ScalaConfigProvider implements vscode.DebugConfigurationProvider {
   }
 }
 
+class ScalaDebugServerFactory implements vscode.DebugAdapterDescriptorFactory {
+  createDebugAdapterDescriptor(
+    session: vscode.DebugSession
+  ): ProviderResult<DebugAdapterDescriptor> {
+    if (
+      session.configuration.mainClass !== undefined ||
+      session.configuration.testClass !== undefined
+    ) {
+      return vscode.commands
+        .executeCommand<DebugSession>(
```

Comment:
Maybe parallel to executing the command we could create the configuration?

Reply:
by this moment configuration is already created and defines what's going on, see `session.configuration`.

Reply:
I was thinking that we should create a configuration when running via code lenses, so that the user can easily retry. Not sure where to put that though.

Reply:
I think, this is good idea, but quite big to make it in another PR.